File: web/domain/design/entities/design_asset.ts
export const DesignAssetType = {
  TEXT: 6,
};

export const DesignAssetPropertyKey = Object.freeze({
  CONTENT: "text",
  FONT_NAME: "font_name",
  FONT_SIZE: "font_size",
});

export interface DesignAssetProps {
  id: number;
  design_id: number;
  project_id: number;
  type: number;
  Width: number;
  Height: number;
  properties?: Property[];
}

interface Property {
  key: string;
  value: string;
}

export class DesignAsset {
  private p: DesignAssetProps;
  constructor(p: DesignAssetProps) {
    this.p = p;
  }

  static create(p: DesignAssetProps): DesignAsset {
    return new DesignAsset(p);
  }

  get id() {
    return this.p.id;
  }

  get desinID(){
    return this.p.design_id
  }

  get type() {
    return this.p.type;
  }

  get text() {
    if (!this.p.properties) return;
    const textProp = this.p.properties.filter(
      (p) => p.key === DesignAssetPropertyKey.CONTENT,
    );
    if (textProp.length === 0) return;
    const text = textProp[0].value;
    return text;
  }

  get fonts(): string[] {
    if (!this.p.properties) return [];
    const fontsName = this.p.properties.filter(
      (p) => p.key === DesignAssetPropertyKey.FONT_NAME,
    );
    return fontsName.map((t) => t.value);
  }

  get fontSize(){
    if (!this.p.properties) return;
    const fontSize = this.p.properties.filter(
      (p) => p.key === DesignAssetPropertyKey.FONT_SIZE,
    );
    if (fontSize.length === 0) return;
    const text = fontSize[0].value;
    return text;
  }
}

File: internal/domain/entities/design_asset.go
package entities

type DesignAsset struct {
	ID            int32                     `json:"id,omitempty"`
	DesignID      int32                     `json:"design_id,omitempty"`
	AlternativeTo int32                     `json:"alternative_to,omitempty"`
	ProjectID     int32                     `json:"project_id,omitempty"`
	Properties    []DesignAssetPropertyData `json:"properties,omitempty"`
	PathURL       string                    `json:"path_url,omitempty"`
	AssetURL      string                    `json:"asset_url,omitempty"`
	Type          DesignAssetType           `json:"type,omitempty"`
	Width         int32                     `json:"width,omitempty"`
	Height        int32                     `json:"height,omitempty"`
}

type DesignAssetPropertyData struct {
	Key   string `json:"key,omitempty"`
	Value string `json:"value,omitempty"`
}
